integration/daemon/daemon_linux_test.go

```diff
@@ -3,43 +3,45 @@ package daemon // import "github.com/docker/docker/integration/daemon"
 import (
 	"context"
 	"net"
+	"net/netip"
 	"testing"
 
 	"github.com/docker/docker/api/types/network"
+	"github.com/docker/docker/internal/nlwrap"
+	"github.com/docker/docker/internal/testutils/networking"
 	"github.com/docker/docker/testutil"
 	"github.com/docker/docker/testutil/daemon"
 	"github.com/vishvananda/netlink"
 	"gotest.tools/v3/assert"
 	is "gotest.tools/v3/assert/cmp"
-	"gotest.tools/v3/icmd"
 	"gotest.tools/v3/skip"
 )
 
+// Check that the daemon will start with fixed-cidr set, but no bip.
+// Regression test for https://github.com/moby/moby/issues/45356
 func TestDaemonDefaultBridgeWithFixedCidrButNoBip(t *testing.T) {
+	skip.If(t, testEnv.IsRootless, "can't create L3 segment in rootless namespace")
 	ctx := testutil.StartSpan(baseContext, t)
 
-	bridgeName := "ext-bridge1"
-	d := daemon.New(t, daemon.WithEnvVars("DOCKER_TEST_CREATE_DEFAULT_BRIDGE="+bridgeName))
-	defer func() {
-		d.Stop(t)
-		d.Cleanup(t)
-	}()
+	host, cleanup := newHostInL3Seg(t, "fcnobip", "192.168.130.1", "fd69:d2cd:f7df::1")
+	defer cleanup()
 
-	defer func() {
-		// No need to clean up when running this test in rootless mode, as the
-		// interface is deleted when the daemon is stopped and the netns
-		// reclaimed by the kernel.
-		if !testEnv.IsRootless() {
-			deleteInterface(t, bridgeName)
-		}
-	}()
-	d.StartWithBusybox(ctx, t, "--bridge", bridgeName, "--fixed-cidr", "192.168.130.0/24")
+	host.Do(t, func() {
+		// Run without OTel because there's no routing from this netns for it - which
+		// means the daemon doesn't shut down cleanly, causing the test to fail.
+		d := daemon.New(t, daemon.WithEnvVars("OTEL_EXPORTER_OTLP_ENDPOINT="))
```

Comment:
Would this be something we could control through the `ctx := testutil.StartSpan(baseContext, t)` utility? It looks like that checks if Otel is configured in the environment; perhaps that could also take an argument to run `t.SetEnv()` for the test (and restore it after). Not  sure if `t.SetEnv` would be problematic if we want to run things in parallel though.


(definitely in favour of having an option for this, so that we don't have to know about specific env-vars everywhere)